raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,697 @@
+pragma solidity ^0.4.17;
+
+import "./Token.sol";
+import "./Utils.sol";
+import "./SecretRegistry.sol";
+
+contract TokenNetwork is Utils {
+
+    /*
+     *  Data structures
+     */
+
+    string constant public contract_version = "0.3._";
+
+    // Instance of the token used as digital currency by the channels
+    Token public token;
+
+    // Instance of SecretRegistry used for storing secrets revealed in a mediating transfer.
+    SecretRegistry public secret_registry;
+
+    // Channel identifier is a uint256, incremented after each new channel
+    mapping (uint256 => Channel) public channels;
+
+    // Used for determining the next channel identifier
+    // Start from 1 instead of 0, otherwise the first channel will have an additional
+    // 15000 gas cost than the rest
+    uint256 public last_channel_index = 1;
+
+    // The ClosingRequest identfier must be THE SAME as the channel identifier (mapping key) from `channels`
+    mapping (uint256 => ClosingRequest) public closing_requests;
```

Comment:
What is the verdict? does it save in storage costs to have a separate mapping?

Reply:
Continuing this here: https://github.com/raiden-network/raiden-contracts/issues/3